ann_benchmarks/datasets.py

```diff
@@ -192,6 +194,33 @@ def fashion_mnist(out_fn):
     test = _load_mnist_vectors('fashion-mnist-test.gz')
     write_output(train, test, out_fn, 'euclidean')
 
+# Creates a 'deep image descriptor' dataset using the 'deep10M.fvecs' sample
+# from http://sites.skoltech.ru/compvision/noimi/. The download logic is adapted
+# from the script https://github.com/arbabenko/GNOIMI/blob/master/downloadDeep1B.py.
+def deep_image(out_fn):
+    train_size = 10 * 1000 * 1000
+    test_size = 1000
+
+    yadisk_key = 'https://yadi.sk/d/11eDCm7Dsn9GA'
+    response = urlopen('https://cloud-api.yandex.net/v1/disk/public/resources/download?public_key=' \
+        + yadisk_key + '&path=/base/base_00')
+    response_body = response.read().decode("utf-8")
+
+    dataset_url = response_body.split(',')[0][9:-1]
+    filename = os.path.join('data', 'deep-image.fvecs')
+    download(dataset_url, filename)
+
+    # In the fvecs file format, each vector is stored by first writing its
+    # length as an integer, then writing its components as floats.
+    fv = numpy.fromfile(filename, dtype=numpy.float32)
+    dim = fv.view(numpy.int32)[0]
+
+    limit = (dim + 1) * (train_size + test_size)
+    fv = fv[0:limit]
+    X_train = fv.reshape(-1, dim + 1)[:, 1:]
+
+    X_train, X_test = train_test_split(X, test_size=test_size)
```

Comment:
the code currently fails on this line since `X` isn't defined. should L220 say `X` rather than `X_train`?